sbin/umount/umount.c

```diff
@@ -455,6 +471,14 @@ umountfs(struct statfs *sfs)
 		clnt_destroy(clp);
 	}
 	free(orignfsdirname);
+
+	if (dflag) {
+		if (md_detach(sfs->f_mntfromname) && !multiple)
+			return (-1);
+		else if (vflag)
```

Comment:
why is this an `else if` and not just an `if`?

Reply:
> why is this an `else if` and not just an `if`?

Fixed the whole block logic.